Linear kepsilon

tanoret opened the pull request:
Closes #29846
Closes #31117

tanoret committed:
initial includes and src implementation for lienar kEpsilon Refs #0

tanoret committed:
adding channel test Refs #000

grmnptr committed:
Register nontime tag just in case, make sure automatic scaling is not taken into account for linear systems.

grmnptr committed:
Register nontime tag just in case, make sure automatic scaling is not taken into account for linear systems.

grmnptr committed:
Correct deviatoric terms in the momentum kernel.

grmnptr committed:
Add gradient refreshers to SNL.

grmnptr committed:
Add tests to verify that we converge to analytic solutions.

grmnptr committed:
Move solution update to the momentum system loop.

tanoret committed:
first working linear k epsilon model Refs #29846

Commit:
adding transients tests Refs #29846

tanoret committed:
adding documentation Refs #29846

tanoret commented:
@grmnptr @GiudGiud - could you please do the code review?
@freiler - could you please do the technical review? I normally run the tests in the flow test bed and was getting the same results than the k-epsilon with the nonlinear assembly

Thanks!!

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2679950/), step Clang format on 81a7048 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 93913d0250cfa598a8b85bbdcaf5be5af83f43bf`

tanoret committed:
correcting style Refs #29846

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2681004/), step Clang format on b590f08 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 5fce0ac136837e85742ff9c1539c66314ba1f773`

lindsayad commented on framework/src/linearfvbcs/LinearFVBoundaryCondition.C:
```diff
@@ -49,7 +49,7 @@ LinearFVBoundaryCondition::LinearFVBoundaryCondition(const InputParameters & par
                            false,
                            "variable",
-                           Moose::VarKindType::VAR_SOLVER,
+                           Moose::VarKindType::VAR_ANY,
```
why this change?

GiudGiud replied:
this is allowing BCs on aux variables. Peter will do a separate PR

grmnptr replied:
Well, this is in this PR:
https://github.com/idaholab/moose/pull/29532

lindsayad commented on framework/src/problems/FEProblemBase.C:
```diff
@@ -7420,22 +7425,26 @@ FEProblemBase::computeLinearSystemTags(const NumericVector<Number> & soln,
     _functions.jacobianSetup(tid);
   }
 
-  try
-  {
-    computeSystems(EXEC_NONLINEAR);
-  }
-  catch (MooseException & e)
-  {
-    _console << "\nA MooseException was raised during Auxiliary variable computation.\n"
-             << "The next solve will fail, the timestep will be reduced, and we will try again.\n"
-             << std::endl;
```
we definitely want try-catch

lindsayad reviewed (commented):
Need some communication about why we are changing the framework objects. Seems to be to allow BCs on aux variables?

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2681304/), step Docs: sync website on 57a1783 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/29847/site)

This comment will be updated on new commits.

GiudGiud commented on modules/navier_stokes/src/linearfvkernels/LinearFVTKESourceSink.C:
```diff
@@ -0,0 +1,269 @@
+    // Assembly variables
+    Real destruction = 0.0;
+    Real tot_weight = 0.0;
+    std::vector<Real> y_plus_vec, velocity_grad_norm_vec;
```
you should reserve a size for these two vectors for performance purposes rather than allocating dynamically later on

tanoret replied:
Thanks. These vectors have one or two components. Reserving should make no difference or even slow down the process. Please advise

GiudGiud commented on modules/navier_stokes/src/linearfvkernels/LinearFVTurbulentLimitedAdvection.C:
```diff
@@ -0,0 +1,179 @@
+
+  // If we are on an internal face, we populate the four entries in the system matrix
+  // which touch the face
+  if (_current_face_type == FaceInfo::VarFaceNeighbors::BOTH)
```
let's discuss this tomorrow. this is a ton code duplication for something that block restriction should achieve naturally?

GiudGiud replied:
if you dont think so, then let's create "addMatrixContrib/ETCInternal in the advection class.

then in the base class (regular advection) you do: addMatrixContrib() { addMatrixContribInternal() }

and in the derived class, you do 
if (on boundary)
  code code
else
 addMatrixContribInternal();

tanoret replied:
I am tabling this for now since this is a bit unrelated to this PR. Let's keep this optimization for future efforts.

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/gold/channel_ERCOFTAC_out.e:
well actually.

I dont think we need the short versions. The long versions run in 4 and 6s respectively in devel mode. this is plenty fast enough imo.

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    v = vel_y
+    momentum_component = 'x'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    type = LinearFVDiffusion
+    variable = vel_x
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    v = vel_y
+    momentum_component = 'y'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    type = LinearFVDiffusion
+    variable = vel_y
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    type = LinearFVAnisotropicDiffusion
+    variable = pressure
+    diffusion_tensor = Ainv
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKE
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    variable = TKE
+    diffusion_coeff = 'mu_t'
+    scaling_coeff = ${sigma_k}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKED
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+    variable = TKED
+    diffusion_coeff = 'mu_t'
+    scaling_coeff = ${sigma_eps}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+[]
+
+[Executioner]
+  type = PIMPLE
```
I think we could put this and the steady version ot the same folder and reuse most of the input files. You can use the `!include mesh.i` syntax etc.

tanoret replied:
I would like to have complete working tests based on the feedback that I have received from our users. The include requires people accessing two files instead of one, which complicates the understanding. Thank you

grmnptr replied:
Fine.

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    v = vel_y
+    momentum_component = 'x'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    type = LinearFVDiffusion
+    variable = vel_x
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    v = vel_y
+    momentum_component = 'y'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    type = LinearFVDiffusion
+    variable = vel_y
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    type = LinearFVAnisotropicDiffusion
+    variable = pressure
+    diffusion_tensor = Ainv
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKE
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    variable = TKE
+    diffusion_coeff = 'mu_t'
+    scaling_coeff = ${sigma_k}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKED
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,417 @@
+    variable = TKED
+    diffusion_coeff = 'mu_t'
+    scaling_coeff = ${sigma_eps}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated/tests:
```diff
@@ -0,0 +1,23 @@
+  design = 'INSFVTurbulentAdvection.md INSFVTurbulentDiffusion.md INSFVTKESourceSink.md INSFVTKEDSourceSink.md INSFVTurbulentViscosityWallFunction.md kEpsilonViscosityAux.md'
+  issues = '#9007'
+  [channel_ERCOFTAC_linear]
+    requirement = 'The system shall be able to solve fluid flow problems with k-epsilon turbulence model for a standard channel, '
```
```suggestion
    requirement = 'The system shall be able to solve fluid flow problems with k-epsilon turbulence model for a standard channel with linear FV discretization, and'
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/segregated/channel_ERCOFTAC.i:
```diff
@@ -10,7 +10,7 @@
 ##########################################################
 
 H = 1 #halfwidth of the channel
```
Part of the reason for regolds is the we fixed the deviatoric term for nonlinear assembly.

grmnptr replied:
If we are already regolding, I think it is better to have the same conditions for linear and nonlinear. Linear can't do symmetry at the moment.

tanoret replied:
Makes sense. Keeping them as is for now

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/segregated/channel_ERCOFTAC.i:
```diff
@@ -123,7 +141,7 @@ wall_treatment = 'eq_newton' # Options: eq_newton, eq_incremental, eq_linearized
     mu = 'mu_t'
     momentum_component = 'x'
-    complete_expansion = true
+    complete_expansion = no
```
the deviatoric part should be fixed now, at least for laminar cases.

tanoret replied:
Yes, but we don't need it for this a channel case. There is no need to test it here

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/segregated/channel_ERCOFTAC.i:
```diff
@@ -272,24 +292,37 @@ wall_treatment = 'eq_newton' # Options: eq_newton, eq_incremental, eq_linearized
-  []
+    value = '${eps_init}'
+  []
+  # [inlet_TKE]
+  #   type = INSFVInletIntensityTKEBC
+  #   boundary = 'left'
+  #   variable = TKE
+  #   u = vel_x
+  #   v = vel_y
+  #   intensity = ${intensity}
+  # []
+  # [inlet_TKED]
+  #   type = INSFVMixingLengthTKEDBC
+  #   boundary = 'left'
+  #   variable = TKED
+  #   k = TKE
+  #   characteristic_length = '${fparse 2*H}'
+  # []
```
```suggestion
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/segregated/channel_ERCOFTAC.i:
```diff
@@ -299,39 +332,39 @@ wall_treatment = 'eq_newton' # Options: eq_newton, eq_incremental, eq_linearized
-    boundary = 'bottom'
-    variable = TKED
-  []
+  # [sym-u]
+  #   type = INSFVSymmetryVelocityBC
+  #   boundary = 'bottom'
+  #   variable = vel_x
+  #   u = vel_x
+  #   v = vel_y
+  #   mu = 'mu_t'
+  #   momentum_component = x
+  # []
+  # [sym-v]
+  #   type = INSFVSymmetryVelocityBC
+  #   boundary = 'bottom'
+  #   variable = vel_y
+  #   u = vel_x
+  #   v = vel_y
+  #   mu = 'mu_t'
+  #   momentum_component = y
+  # []
+  # [symmetry_pressure]
+  #   type = INSFVSymmetryPressureBC
+  #   boundary = 'bottom'
+  #   variable = pressure
+  # []
+  # [symmetry_TKE]
+  #   type = INSFVSymmetryScalarBC
+  #   boundary = 'bottom'
+  #   variable = TKE
+  # []
+  # [symmetry_TKED]
+  #   type = INSFVSymmetryScalarBC
+  #   boundary = 'bottom'
+  #   variable = TKED
+  # []
```
```suggestion
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    variable = TKED
+    diffusion_coeff = 'mu_t'
+    scaling_coeff = ${sigma_eps}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKED
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    type = LinearFVTurbulentLimitedDiffusion
+    variable = TKE
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    type = LinearFVAnisotropicDiffusion
+    variable = pressure
+    diffusion_tensor = Ainv
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    type = LinearFVDiffusion
+    variable = vel_y
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    v = vel_y
+    momentum_component = 'y'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    type = LinearFVDiffusion
+    variable = vel_x
+    diffusion_coeff = ${mu}
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-non-eq-wall.i:
```diff
@@ -0,0 +1,358 @@
+    v = vel_y
+    momentum_component = 'x'
+    rhie_chow_user_object = 'rc'
+    use_nonorthogonal_correction = true
```
```suggestion
    use_nonorthogonal_correction = false
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/lid-driven-turb-std-wall.i:
```diff
@@ -0,0 +1,358 @@
+### Thermophysical Properties ###
```
We only have a few options different in this, we should either do the `inlude` syntax and split the input file or just the cli_args.

tanoret replied:
We have it the same way in the nonlinear. I see the interest in spelling options clearly and fully in one test. Keeping it as is for now

tanoret replied:
I want complete test cases, because these are the cases from which people start from. I would prefer to keep this like it is. This is not a mistake, it is a conscious design decision based on feedback from users

GiudGiud replied:
Users should use the Physics syntax. I don't see the need for complete inputs using the old syntax.

I dont want to be debugging 400 line input files because someone was inconsistent between their interpolation method, wall function, or wrote v = u instead of v = v

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/tests:
```diff
@@ -0,0 +1,46 @@
+  design = 'INSFVTurbulentAdvection.md INSFVTurbulentDiffusion.md INSFVTKESourceSink.md INSFVTKEDSourceSink.md INSFVTurbulentViscosityWallFunction.md kEpsilonViscosityAux.md'
+  issues = '#9007'
+  [lid_driven_turb_std_wall_linear]
+    requirement = 'The system shall be able to run fluid flow with k-epsilon turbulence model in enclosures with standard wall functions, '
```
```suggestion
    requirement = 'The system shall be able to run fluid flow with k-epsilon turbulence model in enclosures with standard wall functions and the linear finite volume discretization, '
```

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/linear-segregated/tests:
```diff
@@ -0,0 +1,46 @@
+    []
+  []
+  [lid_driven_turb_non_eq_wall_linear]
+    requirement = 'The system shall be able to run fluid flow with k-epsilon turbulence model in enclosures with non-equilibrium wall functions, '
```
```suggestion
    requirement = 'The system shall be able to run fluid flow with k-epsilon turbulence model in enclosures with non-equilibrium wall functions and the linear finite volume discretization, '
```

grmnptr reviewed (changes requested):
Very nice, just a few suggestions!

tanoret committed:
adding mods for computing efficiency Refs #29846

tanoret commented on framework/src/systems/LinearSystem.C:
```diff
@@ -91,6 +94,10 @@ LinearSystem::LinearSystem(FEProblemBase & fe_problem, const std::string & name)
 
+  // We add other wector tags so that objects acting on the aux system inheriting
+  // from the tagging interface can still be used without any nonlinear systems
+  _rhs_non_time_tag = _fe_problem.addVectorTag("NONTIME");
```
Kept the first one.

tanoret commented on modules/navier_stokes/doc/content/source/linearfvkernels/LinearFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,113 @@
+# LinearFVTKEDSourceSink
+
+This kernel implements a version of [`INSFVTKEDSourceSink`](INSFVTKEDSourceSink.md) for a linear kernel.
+The documentation is repeated down here for completeness.
```
We will eventually be removing the one from the nonlinear assembly. This is why I copied it here. Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/linearfvkernels/LinearFVTurbulentLimitedAdvection.md:
```diff
@@ -0,0 +1,19 @@
+the [!param](/LinearFVKernels/LinearFVTurbulentLimitedAdvection/walls) list.
+For any element that is in contact with a boundary identified
+in the [!param](/LinearFVKernels/LinearFVTurbulentLimitedAdvection/walls) list,
+advection will be skipped for that element over all faces.
```
Thanks. Yes. Keeping it this way for now. May be able to do this in the mesh later, but this is good for now.

tanoret commented on modules/navier_stokes/include/base/NavierStokesMethods.h:
```diff
@@ -99,7 +99,8 @@ Real prandtlPropertyDerivative(const Real & mu,
  * @return the velocity at the wall
  */
-ADReal findUStar(const ADReal & mu, const ADReal & rho, const ADReal & u, Real dist);
+template <typename T>
+T findUStar(const T & mu, const T & rho, const T & u, Real dist);
```
Thanks

tanoret commented on modules/navier_stokes/include/linearfvkernels/LinearFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,85 @@
+  std::map<const Elem *, bool> _wall_bounded;
+  std::map<const Elem *, std::vector<Real>> _dist;
+  std::map<const Elem *, std::vector<const FaceInfo *>> _face_infos;
+  ///@}
```
Thanks. The problem is that INSFVTKED will eventually be going away from the linear methods when we remove the nonlinear assembly. I prefer to keep it this way for now. We can eventually optimize it in the future

tanoret commented on modules/navier_stokes/src/auxkernels/RANSYPlusAux.C:
```diff
@@ -95,16 +95,23 @@ RANSYPlusAux::computeValue()
     for (unsigned int i = 0; i < distance_vec.size(); i++)
     {
-      const auto parallel_speed = NS::computeSpeed(
+      const auto parallel_speed = NS::computeSpeed<ADReal>(
```
Sometimes this wall will be called with ADReals when using the nonlinear methods. Real here will break a bunch of things

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -127,32 +130,111 @@ findyPlus(const ADReal & mu, const ADReal & rho, const ADReal & u, const Real di
+template <typename T>
+T
+computeShearStrainRateNormSquared(const Moose::Functor<T> & u,
+                                  const Moose::Functor<T> * v,
```
Thanks!

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/mms/linear-segregated/2d-vortex/spacedependent_mu/test.py:
```diff
@@ -0,0 +1,117 @@
+            if key in velocity_labels:
+                self.assertTrue(fuzzyAbsoluteEqual(value, 2.0, .4))
+            else:
+                self.assertTrue(fuzzyAbsoluteEqual(value, 1.0, .5))
```
Yeah. We should really not expect 2nd order for any of them in the current formulations...

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/mms/linear-segregated/2d-vortex/spacedependent_mu/test.py:
```diff
@@ -0,0 +1,117 @@
+            if key in velocity_labels:
+                self.assertTrue(fuzzyAbsoluteEqual(value, 2.0, .4))
+            else:
+                self.assertTrue(fuzzyAbsoluteEqual(value, 1.0, .5))
```
Resolving this one.

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/bfs/linear-segregated-transient/tests:
```diff
@@ -0,0 +1,25 @@
+  issues = '#9007'
+  [BFS_ERCOFTAC_linear_transient]
+    requirement = 'The system shall be able to solve fluid flow problems with k-epsilon turbulence model for a backward facing step, '
+    [result]
+      type = 'Exodiff'
+      input = 'BFS_ERCOFTAC.i'
+      exodiff = 'BFS_ERCOFTAC_exodus.e'
+      detail = 'and reach converged results with segregated solvers,'
+      abs_zero = 5e-6
+      rel_err = 5e-6
+      heavy = true
+    []
```
If we have drift in the wall function shear (for non-regression), we are unlikely to see this in the first iterations. We need this longer running tests unfortunately....

GiudGiud replied:
Addressed by running the longer one. Lienar FV is fast enough imo

tanoret committed:
Apply suggestions from code review - first simple round Refs #29846

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>
Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

tanoret commented on modules/navier_stokes/src/linearfvkernels/LinearFVTKESourceSink.C:
```diff
@@ -0,0 +1,269 @@
+  params.addParam<MooseEnum>("wall_treatment",
+                             wall_treatment,
+                             "The method used for computing the wall functions "
+                             "'eq_newton', 'eq_incremental', 'eq_linearized', 'neq'");
```
Non-equilibrium uses one treatment here, all the rest use the others. We need to preserve all names for consistency with other kernels as mu_t will change regarding each of these methods

tanoret replied:
No. For `eq_newton` `eq_incremental` `eq_linearized` one uses the linear treatment to find `yPlus`, while they then defer in the way in which the turbulent viscosity is computed (kEpsilonViscosityAux). This allows us to have a universal selection of wall models for all kernels, which greatly simplify the code. This suggestion is not useful

tanoret replied:
Also, this is already explained in the documentation

tanoret committed:
adding updated and checked test suite for turbulence cases Refs #29846

grmnptr committed:
Update turbulence petsc option parsing.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2729483/), step Clang format on c896aef wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format e6f742bded1f28a5d4fa5991eff1b7008b3a7d2d`

moosebuild commented:
All [jobs](https://civet.inl.gov/event/221546/) on cbd8ef1 : invalidated by @grmnptr

Invalidating all jobs to check which ones are related

tanoret commented on modules/navier_stokes/src/executioners/SIMPLESolveBase.C:
```diff
@@ -27,6 +27,8 @@ SIMPLESolveBase::validParams()
   params.addParam<std::vector<SolverSystemName>>(
       "passive_scalar_systems", {}, "The solver system for each scalar advection equation.");
+  params.addParam<std::vector<SolverSystemName>>(
+      "turbulence_systems", {}, "The solver system for each surrogate turbulence equation.");
```
This will be deleted later on. Better to keep like this for now imo

tanoret commented on modules/navier_stokes/src/executioners/SIMPLESolveBase.C:
```diff
@@ -380,6 +433,44 @@ SIMPLESolveBase::SIMPLESolveBase(Executioner & ex)
                                   "passive_scalar_absolute_tolerance"},
                                  false);
+
+  // We check for input errors with regards to the surrogate turbulence equations. At the same time,
```
Because they are surrogate variables (k-epsilon) that define the TKE and TKED

tanoret committed:
adding guillaume review Refs #29846

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/bfs/linear-segregated-transient/BFS_ERCOFTAC.i:
```diff
@@ -0,0 +1,437 @@
+[]
+
+[Executioner]
+  type = PIMPLE
```
Yes, but I am already testing that. I want to test transients

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/bfs/linear-segregated-transient/BFS_ERCOFTAC.i:
```diff
@@ -0,0 +1,437 @@
+Re = 5100
```
This one is in the `linear-segreagated` folder one directory above

tanoret replied:
I am not doing this. Both transient and steady cases test different phenomena and should not be merged. I will keep the folders as they are. Thank you for the suggestion

grmnptr replied:
Fine.

tanoret committed:
address peter and guillaume review round 2 #29846

tanoret committed:
regolding 2d-momentum-friction due to small difference in y velocity Refs #29846
Due to changes in when solution is updated in nonlinear-SIMPLE

tanoret committed:
correcting includes in header, test headers, and deactivting recover and multi-threading for testing Refs #29846

tanoret committed:
Update test spec to capabilities

grmnptr reviewed (changes requested):
Thanks for pushing this large PR, I think it is getting really close! 
I have a few more things here though. If you are running short on time, I can help with it.

tanoret commented on framework/src/systems/LinearSystem.C:
```diff
@@ -107,6 +114,32 @@ LinearSystem::initialSetup()
+  // Calling initial setpup for the linear kernels
+  // Note: if we ever create a kernel not inheriting from LinearFVElementalKernel
+  // or LinearFVFluxKernel, we will need to add the initialsetup here
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); tid++)
+  {
+    std::vector<LinearFVElementalKernel *> fv_elemental_kernels;
+    _fe_problem.theWarehouse()
+        .query()
+        .template condition<AttribSystem>("LinearFVElementalKernel")
+        .template condition<AttribThread>(tid)
+        .queryInto(fv_elemental_kernels);
+
+    for (auto * fv_kernel : fv_elemental_kernels)
+      fv_kernel->initialSetup();
+
+    std::vector<LinearFVFluxKernel *> fv_flux_kernels;
+    _fe_problem.theWarehouse()
+        .query()
+        .template condition<AttribSystem>("LinearFVFluxKernel")
+        .template condition<AttribThread>(tid)
+        .queryInto(fv_flux_kernels);
+
+    for (auto * fv_kernel : fv_flux_kernels)
+      fv_kernel->initialSetup();
+  }
```
I don't know why to do this or how this would improve the code. I can do the research in the code, but why is this effort worth it?

grmnptr replied:
Similarly to the regular `FVKernels`, we register the attribute a step lower, so the lookup with `LinearFVKernel` would not return anything.

tanoret commented on modules/navier_stokes/src/linearfvkernels/LinearFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,260 @@
+      _wall_boundary_names, _fe_problem, _subproblem, blockIDs(), _wall_bounded);
+}
+
+Real
+LinearFVTurbulentDiffusion::computeElemMatrixContribution()
+{
+  const auto face_arg = makeCDFace(*_current_face_info);
+  return computeFluxMatrixContribution() / _scaling_coeff(face_arg, determineState());
+}
```
I cannot do this because the boundary conditions (near wall functions) do not propagate properly when defining the effective turbulent viscosity from the turbulent viscosity, which have the wall function. I am keeping this like this for now

tanoret committed:
adding reound 2 review from peter Refs #29846

freiler commented:
Channel DNS Re_tau 590 comparison vs CFD benchmark code 
Velocity Profile:
<img width="411" alt="image" src="https://github.com/user-attachments/assets/fa34bfd9-2e08-482c-bef9-b14419a66fc4" />

TKE profile vs nonlinearSIMPLE and CFD benchmark (red):
<img width="474" alt="image" src="https://github.com/user-attachments/assets/4d99e4fb-6902-4f6e-9fea-3ac77d75787c" />

freiler commented:
Backward Facing Step comparison:

![image](https://github.com/user-attachments/assets/338b62d3-1897-4cde-a934-7a855d81f362)

Skin friction coefficient:

<img width="299" alt="image" src="https://github.com/user-attachments/assets/02d511ce-e338-41a7-8c08-550e79662b89" />

freiler commented:
There are two things to analyze which are 

1-Overdiffusion in TKE using non-equilibrium wall functions in the BFS case.
2-Velocity oscillations at the outlet of geometries (see picture below)

![image](https://github.com/user-attachments/assets/30ef8eb4-1626-4346-9891-31e78471214a)

GiudGiud commented:
> 2-Velocity oscillations at the outlet of geometries (see picture below) using the nonlinear model

the nonlinear FV? Are these oscillations in the new tests? If not we can create an issue for it.
I think we have some tests where we just use a dirichletBC on k/eps when outflow would be better

grmnptr commented:
> > 2-Velocity oscillations at the outlet of geometries (see picture below) using the nonlinear model
> 
> the nonlinear FV? Are these oscillations in the new tests? If not we can create an issue for it. I think we have some tests where we just use a dirichletBC on k/eps when outflow would be better

Yeah it is the NL k-epsilon and this behavior got introduced in this PR, devel is fine. I think the main reason is that we added modifications to the full expansion of the stress tensor (should have been a fix).

tanoret committed:
correcting variable state in INSFVTurbulentViscosityWallFunction for lienar mdoel Refs #29846

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2831032/), step Clang format on 4800a8d wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/29847/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 41d1ea116237da6f44c9f9797a03defa91b9f539`

tanoret committed:
lagging limiters #29846

tanoret committed:
adding linear BC for mut and correcting states of durbin limiter Refs #29846

tanoret committed:
regolding tests with the BCs for mu_t  Refs #29846

tanoret committed:
correcting test requirements for sqa Refs #29846

tanoret committed:
removing shearstressaux (not needed until k omega sst comes in) Refs #29846

tanoret committed:
addressing issues with missing doco and application registry Refs #29846

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2978268/), step Generate coverage on 1dcda8c wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>b99593</th>
      <th colspan=4>#29847 1dcda8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.49%</td>
      <td>85.50%</td>
      <td>+0.01%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>114749</td>
      <td>114780</td>
      <td>+31</td>
      <td>18</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19479</td>
      <td>19466</td>
      <td>-13</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage/framework)


## Modules coverage

### Fluid properties

<table>
  <thead>
    <tr>
      <th></th>
      <th>b99593</th>
      <th colspan=4>#29847 1dcda8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.92%</td>
      <td>85.93%</td>
      <td>+0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>8181</td>
      <td>8182</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1341</td>
      <td>1340</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage_diff/fluid_properties)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage/fluid_properties)

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>b99593</th>
      <th colspan=4>#29847 1dcda8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.19%</td>
      <td>85.33%</td>
      <td>+0.14%</td>
      <td>87.38%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>18788</td>
      <td>19314</td>
      <td>+526</td>
      <td>547</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3267</td>
      <td>3321</td>
      <td>+54</td>
      <td>79</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/29847/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/29847/coverage/xfem)

</details>

## Warnings

- `navier_stokes` new line coverage rate 87.38% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud reviewed (dismissed):
I think we may want to discuss taking care of these changes for you @tanoret

what's the reason behind re-golding all the entire nonlinear assembly turbulence cases?
is it the missing deviatoric term? Should we just set those cases to not use the deviatoric term then?

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/bfs/linear-segregated/tests:
```diff
@@ -0,0 +1,27 @@
+  design = 'LinearFVTurbulentAdvection.md LinearFVTurbulentDiffusion.md LinearFVTKESourceSink.md LinearFVTKEDSourceSink.md kEpsilonViscosityAux.md INSFVTurbulentViscosityWallFunction.md'
+  issues = '#29846'
+  [BFS_ERCOFTAC_linear]
+    requirement = 'The system shall be able to solve fluid flow problems with the linear k-epsilon turbulence model for a backward facing step, '
```
```suggestion
    requirement = 'The system shall be able to solve steady-state fluid flow problems with the linear k-epsilon turbulence model for a backward facing step, '
```

grmnptr reviewed (changes requested):
Okay, this is getting really close!
Based on yesterday's debugging sesssion I think we should add one more test case (I can add it this weekend) not physical, just regression:
A channel:
```
| fluid | solid  |
|   ^    |           |
|   ^    |           |
|   ^    |           |
|   ^    |           |
```
Solid can be blank. I expect this to work considering our default elemen/face info ordering. Then It would be nice to do the opposite:

```
| solid | fluid  |
|          |   ^    |
|          |   ^    |
|          |   ^    |
|          |   ^    |
```
In this case, the element-side of face info will not be on the fluid domain at the boundary.
This is just to test that the algorithms fetching the wall-bounded elements/sides etc work with block restriction.

grmnptr committed:
Remove bool from wall bounded maps. (#29846)

grmnptr committed:
Make YplusAux nonad. (#29846)

grmnptr committed:
Remove parameters from NL assembly. (#29846)

grmnptr commented on modules/navier_stokes/src/executioners/SIMPLESolveNonlinearAssembly.C:
```diff
@@ -308,12 +308,10 @@ SIMPLESolveNonlinearAssembly::solveMomentumPredictor()
-        libMesh::cast_ref<NonlinearImplicitSystem &>(_momentum_systems[system_i]->system());
+    // We clear the gradient caches of the velocity variables since the deviatoric terms will
+    // need the updated cell gradients. This will also help the turbulence routines which will
+    // need the latest cell gradients.
```
Dangling comment from a modified line. Well I added something here which got removed later.

grmnptr committed:
Add error messages in case of someone wants to build a system for nut. (#29846)

grmnptr committed:
Add subdomain checks to the routines fetching wall distances. (#29846)

grmnptr committed:
Correct block-restriction for deviatoric term and wall function BC. (#29846)

grmnptr committed:
Add test for blockrestricted case. (#29846)

GiudGiud committed:
Fix indexing into residual vector

freiler committed:
Modify channel test, addition of 1 radial cell for enhanced convergence

grmnptr committed:
Restore correct indexing for turbulence and active scalars.

tanoret committed:
fixing channel test cases Refs #29846

grmnptr committed:
Fix block-restriction issues with Moose-LinearFV-NS.

grmnptr committed:
Add modifications to framework kernels and regold cht problem. (#31117)

GiudGiud committed:
Request velocity cell gradients for TKE/D terms

GiudGiud committed:
Save relaxed and limited solution as old solution
when using relaxation

GiudGiud committed:
Move previously regolded files to the new folders of the tests, prepare for conversion to csvdiff

GiudGiud committed:
Remove deprecation failures from MMS heavy test

GiudGiud committed:
Convert segregated BFS test case to CSVDiff

GiudGiud committed:
Convert linear-segregated BFS test case to CSVDiff

GiudGiud committed:
Convert transient linear-segregated BFS test case to CSVDiff
Remove heavy & debug separate testing as test runs fast enough for regular testing

GiudGiud committed:
Convert segregated channel case to CSVDiff
Remove heavy test as test only takes 2.4s to run

GiudGiud committed:
Convert linear-segregated ercoftac channel case to CSVDiff
Remove heavy test as test only takes 1.1s to run

GiudGiud committed:
Convert transient linear-segregated channel case to CSVDiff
Only test in opt mode as takes 3s to run

GiudGiud committed:
Prevent distributed mesh failures in sampler-based CSVdiff testing with element-id
parallel numbering differences with distributed meshes

GiudGiud committed:
Convert lid driven linear-segregated tests to CSVDiff
Only test in heavy as takes 4s to run

GiudGiud committed:
Convert lid-driven newton tests to CSVDiff

GiudGiud committed:
Convert lid-driven Newton + Physics to CSVDiff

GiudGiud committed:
Convert lid driven nonlinearFV + SIMPLE testing to CSVDiffs

GiudGiud committed:
Document default turbulent minimum field value

GiudGiud committed:
Transition turbulence block restriction test to CSVDiff

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/bfs/linear-segregated-transient/tests:
```diff
@@ -0,0 +1,29 @@
+      cli_args = 'Executioner/dt=1.0 Executioner/num_steps=1 Outputs/exodus/file_base=BFS_ERCOFTAC_short'
+      recover = false # we don't support recovery for PIMPLE yet
+      max_threads = 1 # see libmesh issue #3808
+    []
```
only have result, it's quite fast

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/channel/linear-segregated-transient/channel_ERCOFTAC.i:
```diff
@@ -0,0 +1,440 @@
+H = 1 #halfwidth of the channel
```
it is the half-width in this case. leaving as is

GiudGiud committed:
Address missed comments in the documentation & spatial mu tests

GiudGiud committed:
Turn off 2-term BC exapnsion for k-epsilon in Physics by default`
Turn off 2-term BC expansion for k-epsilon Jacobian testing for consistency

grmnptr reviewed (approved):
Looks good to me!
